CompetitionTemplate/src/competition/subsystems/autonomous/DriveForDistanceCommand.java

```diff
@@ -0,0 +1,60 @@
+package competition.subsystems.autonomous;
+
+import com.google.inject.Inject;
+
+import competition.subsystems.drive.DriveSubsystem;
+import competition.subsystems.drive.PoseSubsystem;
+import xbot.common.command.BaseCommand;
+import xbot.common.math.PIDManager;
+import xbot.common.properties.DoubleProperty;
+import xbot.common.properties.XPropertyManager;
+
+public class DriveForDistanceCommand extends BaseCommand{
+    DriveSubsystem driveSubsystem; 
+    PoseSubsystem poseSubsystem;
+    
+    DoubleProperty distancePerTick;
+    DoubleProperty targetRange;
+    
+    PIDManager distanceDrivePid;
+    
+    public double targetDistance;
+    double targetEncoderDistance;
+    double currentDistance;
+    
+    @Inject
+    public DriveForDistanceCommand(PoseSubsystem poseSubsystem, DriveSubsystem driveSubsystem, XPropertyManager propManager){
+        this.driveSubsystem = driveSubsystem;
+        this.poseSubsystem = poseSubsystem;
+        
+        distancePerTick = propManager.createPersistentProperty("the distance per tick on encoder", 10.0);
```

Comment:
I'm confused about why you need this, shouldn't you just be working on the distance given by the pose subystem? You shouldn't know anything about the encoders underneath.